basic-javascript/project-1.spec.js

```diff
@@ -1,3 +1,317 @@
+    const even = helpers.isEven;
+
+    const result1 = even(12);
+    const result2 = even(11);
+
+    expect(result1).toBe(true);
+    expect(result2).toBe(false);
+})
+
+it('should check if a number is odd',() => {
+    const odd = helpers.isOdd
+
+    const result1 = odd(12);
+    const result2 = odd(11);
+
+    expect(result1).toBe(false);
+    expect(result2).toBe(true);
+})
+
+it('should return the square of the number',() => {
+    const square = helpers.square;
+
+    const result1 = square(3);
+    const result2 = square(-3);
+
+    expect(result1).toBe(9);
+    expect(result2).toBe(9);
+})
+
+it('should return the cube of the number',() => {
+    const cube = helpers.cube;
+
+    const result1 = cube(3);
+    const result2 = cube(-3);
+
+    expect(result1).toBe(27);
+    expect(result2).toBe(-27);
+})
+
+it('should raise the number to the exponent provided',() => {
+    const power = helpers.raiseToPower;
+
+    const result1 = power(5,2)
+    const result2 = power(5,-2)
+
+    expect(result1).toBe(25);
+    expect(result2).toBe(.04);
+})
+
+it('round number',() => {
+    const round = helpers.roundNumber;
+
+    const result1 = round(4.2)
+    const result2 = round(4.6)
+
+    expect(result1).toBe(4);
+    expect(result2).toBe(5);
+})
+
+it('round number up',() => {
+    const roundUp = helpers.roundUp;
+
+    const result1 = roundUp(4.2)
+    const result2 = roundUp(4.6)
+
+    expect(result1).toBe(5);
+    expect(result2).toBe(5);
+})
+
+
+it('should add an exclamation point',() => {
+    const addEx = helpers.addExclamationPoint;
+
+    const result1 = addEx('this is a string');
+    const result2 = addEx();
+
+    expect(result1).toBe('this is a string!');
+    expect(result2).toBe(false);
+}) 
+
+it('should combine names',() => {
+    const combine = helpers.combineNames;
+
+    const result1 = combine('firstName','lastName')
+    const result2 = combine();
+
+    expect(result1).toBe('firstName lastName');
+    expect(result2).toBe(false);
+}) 
+
+it('should greet using provided name',() => {
+    const greeting = helpers.getGreeting;
+
+    const result1 = greeting('person');
+    const result2 = greeting();
+
+    expect(result1).toBe('Hello person!');
+    expect(result2).toBe('Hello!');
+}) 
+
+it('should return the rectangle area',() => {
+    const recArea = helpers.getRectangleArea;
+
+    const result1 = recArea(3,4);
+    const result2 = recArea(3,-4)
+
+    expect(result1).toBe(12);
+    expect(result2).toBe(-12);
+}) 
+
+it('should return the triangle area',() => {
+    const triArea = helpers.getTriangleArea;
+
+    const result1 = triArea(3,4);
+    const result2 = triArea(3,-4)
+
+    expect(result1).toBe(6);
+    expect(result2).toBe(-6);
+}) 
+
+it('should return the circle area',() => {
+    const circArea = helpers.getCircleArea;
+
+    const result1 = circArea(1);
+    const result2 = circArea(5)
+
+    expect(result1).toBeCloseTo(3.14,1);
+    expect(result2).toBeCloseTo(78.54,1);
+}) 
```

Comment:
This is a great one to use when dealing with irrational numbers like pi because then you don't have to remember exactly how many decimal places js is going to calculate everything to.